src/pamiq_core/model/interface.py

```diff
@@ -86,3 +86,34 @@ def forward(self, *args, **kwds) -> Any:
     def __call__(self, *args, **kwds) -> Any:
         """Calls `forward` method."""
         return self.forward(*args, **kwds)
+
+    def sync(self) -> None:
+        """Synchronizes parameters of training model to self._inference_model
+        if needed."""
+        if self._need_sync:
+            self.sync_impl(self.inference_model)
+
+    @property
+    def _need_sync(self) -> bool:
+        """Return whether It is necessary to synchronize training model and
+        inference model."""
+        return self.has_inference_model and (not self.inference_only)
+
+    def sync_impl(self, inference_model: T) -> None:
+        """Copies params of training model to self._inference_model if needed.
+
+        Args:
+            T: InferenceModel to sync.
+
+        Example:
+            class TrainingTorchModel(TrainingModel[InferenceTorchModel]):
+                @override
+                def sync_impl(self, inference_model: InferenceTorchModel) -> None:
+                    with torch.no_grad():
+                        for inference_param, training_param in zip(
+                            inference_model.model.parameters(),
+                            self.model.parameters(),
+                        ):
+                            inference_param.copy_(training_param)
+        """
+        raise NotImplementedError
```

Comment:
このリポジトリのコーディング規約に則り、原則としてdoc stringの中に型情報は書かない。Exampleは下手なバイアスを実装者に与えてしまうため、書かない方向性でお願いしたい。
```suggestion
    def sync_impl(self, inference_model: T) -> None:
        """Copies params of training model to self._inference_model if needed.

        Args:
            InferenceModel to sync.
        """
        raise NotImplementedError
```